Docker.DotNet.BasicAuth/BasicAuthCredentials.cs

```diff
@@ -1,18 +1,68 @@
 ﻿using System;
 using System.Net.Http;
+using System.Net.Http.Headers;
 using System.Runtime.InteropServices;
 using System.Security;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Docker.DotNet.BasicAuth
 {
     public class BasicAuthCredentials : Credentials
     {
-        private readonly HttpMessageHandler _handler;
-        private readonly SecureString _username;
-        private readonly SecureString _password;
+        private class BasicAuthHandler : HttpClientHandler
```

Comment:
any specific reason why we're embedding this `BasicAuthHandler ` class inside `BasicAuthCredentials` class? If not, let's move out.